device/common/include/traccc/seeding/device/weight_updating.hpp

```diff
@@ -0,0 +1,34 @@
+// Project include(s).
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/device/get_prefix_sum.hpp"
+#include "traccc/edm/device/triplet_counter.hpp"
+#include "traccc/seeding/detail/seeding_config.hpp"
+#include "traccc/seeding/detail/spacepoint_grid.hpp"
+#include "traccc/seeding/detail/triplet.hpp"
+
+// System include(s).
+#include <cassert>
+#include <cstddef>
+
+namespace traccc::device {
+
+/// Function used for updating the triplets' weights
+///
+/// @param[in] globalIndex   The index of the current thread
+/// @param[in] filter_config        Seedfilter configuration
+/// @param[in] sp_view       The spacepoint grid
+/// @param[in] triplet_ps_view    Prefix sum for iterating over the triplets
+/// @param[out] triplet_view Container storing the triplets
+///
+TRACCC_HOST_DEVICE
+void update_weights(const std::size_t globalIndex,
```

Comment:
The function name is okay like this. Though I would prefer to make it even a bit more explicit, and call it `traccc::device::update_triplet_weights`.

But importantly, make sure that the file names correspond to the function names. At first I thought the function would be called `traccc::device::weight_updating`...